src/generators/legacy-json/types.d.ts

```diff
@@ -0,0 +1,83 @@
+import { ListItem } from 'mdast';
+
+export interface HierarchizedEntry extends ApiDocMetadataEntry {
+  hierarchyChildren: Array<ApiDocMetadataEntry>;
+}
+
+export interface Meta {
```

Comment:
I would love it if you could document a bit these types; What each property means and examples (for all these types)